pkg/trace/stats/aggregation.go

```diff
@@ -80,23 +88,61 @@ func NewAggregationFromSpan(s *pb.Span, origin string, aggKey PayloadAggregation
 			Synthetics: synthetics,
 		},
 	}
-	if enablePeerSvcAgg {
-		agg.PeerService = s.Meta[tagPeerService]
+	var peerTags []string
+	if clientOrProducer(agg.SpanKind) {
+		if enablePeerSvcAgg {
+			agg.PeerService = s.Meta[tagPeerService]
+		}
+		peerTags = matchingPeerTags(s, peerTagKeys)
+		agg.PeerTagsHash = peerTagsHash(peerTags)
 	}
-	return agg
+	return agg, peerTags
+}
+
+func matchingPeerTags(s *pb.Span, peerTagKeys []string) []string {
+	if len(peerTagKeys) == 0 {
+		return nil
+	}
+	var pt []string
+	for _, t := range peerTagKeys {
+		if v, ok := s.Meta[t]; ok {
+			pt = append(pt, t+":"+v)
+		}
+	}
+	return pt
+}
+
+func peerTagsHash(tags []string) uint64 {
+	if len(tags) == 0 {
+		return 0
+	}
+	if !sort.StringsAreSorted(tags) {
+		sort.Strings(tags)
+	}
+	h := fnv.New64a()
+	for i, t := range tags {
+		h.Write([]byte(t))
+		if i != len(tags)-1 {
+			h.Write([]byte{0})
+		}
```

Comment:
```suggestion
		if i > 0 {
			h.Write([]byte{0})
		}
		h.Write([]byte(t))
```
slightly simpler alternative